### PR TITLE
Space Dragon Update: Up Close and Personal

### PR DESCRIPTION
**Pull request by IndieanaJones**
## About The Pull Request

This PR seeks to shore up some weaknesses with the current Space Dragon design to make the experience more interesting for both sides overall, specifically for the dragon.

**WHAT'S UP WITH DRAGON:**

- According to the antagonist poll, dragon is considered in the middle of the midround rankings of being fun to fight. Not terrible!
- However, dragon was 3rd lowest on being fun to play as, so that should be looked into.

Some personal thoughts on why this is:

- Dragon's main draw is the ability to play as a big monster and terrorize the crew. However, Space Dragon's past nerfs have put him to a point where this level of empowerment was greatly diminished.
- One of the things people like Space Dragon for is being able to spawn as a carp when you die to assist the dragon, as it brings in ghosts back into the round in an antagonist role. However, carps are easily killed by Space Dragon's fire breath, and lack enough health to properly defend the rift. Therefore, they usually are forced to go sabotage the station instead of hang around the dragon, who ultimately is a major safety risk to them.
- As it is currently, Space Dragon is weak on most portions of the station, but very strong in certain key areas (AI Satellite, anywhere that's easy to vent). Dragons who utilize these cheesy areas will usually win, other dragons will usually lose.

**THE IDEA BEHIND THE CHANGES:**

- Nerf the cheesy spots dragon can use and force the dragon into a more confrontational position with the crew. Ideally lower the requirements to engage SD consistently enough so more people can actively engage with the antagonist.
- In order to support forcing SD out of the cheese spots and into riskier territory, re-buff SD to feel more like a powerhouse who can compete against the crew.
- Add synergy to Space Dragon and the carps so that carps feel encouraged to stay by the rift and actually help SD with the main objective.

**THE CHANGES:**

- Space Dragon now receives 5 pre-selected locations where their rift can spawn instead of being able to choose wherever for all 3. The locations chosen are specifically risky and in guarded areas of the station. The dragon still only requires 3 rifts to reach its "win-state", and any combination of the 5 locations can be used. Something to note about these locations is that they're usually in the station's interior, so venting these locations will be difficult in most cases.

- Space dragon's health has been increased back back up to the pre-nerf value of 400, from 320
- Fire breath has been sped up to pre-nerf speed (50% faster)
- Space dragon's gust attack takes 20% less time to activate
- Space dragon's gust attack requires LOS to activate

- Carp rifts have a gravity zone around them, meaning gravity is applied normally to anyone within 15 tiles of the rifts. Buffs normal crew movement when gravity is down, nerfs using jetpacks to harass the rift. Being affected by the gravity aura can also tip you off to a nearby rift.
- Carp rift healing aura is now a 3x3 around the rift instead of a 1x1, so more carp can heal at the same time and re-engage in the action
- Carp rift spawn times have been reduced (60 seconds to 45 seconds, not what it was pre-nerf but in-between old and current values)
- Carp rifts are now solid as to prevent characters from hiding underneath them to avoid being hit.

- Space carp spawned from the rift are of a different mob subtype, dubbed "advanced space carp".
- Advanced space carp have 40 health, but only 15 object damage
- Space Dragon's fire breath no longer harms carp, but instead gives them an 8-second speed boost, equivalent to the one Space Dragon gets when enraged. Cold-blooded and everything, it makes sense.

- Space Dragon and rift carp spawn now have their own communication "channel" in the same vein of Blob and its minions.

**SOME TECHIE STUFF**
- Added a gravity aura component. Forces a level of gravity in a radius around the thing that has the component. Only carp rifts use it currently.

## Why It's Good For The Game

Space Dragon has been technically been sitting in a "balanced enough" position for a while now, but optimal space dragon as it is currently is sort of a bore to fight as and against. These changes serve to make SD more exciting to play by forcing it into the station to place rifts while also increasing its strength in order to match the challenge. The changes also make space carp gameplay more about working with the dragon to defend the rifts with actual synergy between the two roles while discouraging going off by yourself to sabotage random stuff. Also gives better communication to the carps and the dragon so they can relay information to each other more reliably.

## Changelog
:cl:
balance: Space Dragon can no longer choose its rift locations freely, and instead is given 5 pre-determined locations to pick from instead
balance: Space Dragon itself has been buffed in order to support a more confrontational playstyle, however its wing gust now requires a line of sight to targets in order to affect them.
balance: Player Space Carp from rifts now have buffed health, but reduced object damage values. They also gain a temporary speed boost when hit by Space Dragon's fire breath instead of taking damage.
balance: Carp rift spawn times have been reduced, the healing AOE is now a 3x3 instead of a 1x1, and apply normal gravity in a large radius around them
balance: Space Dragon and rift carps now communicate on a private channel akin to Blob Overminds and their blob minions.
/:cl:

### CODE REVIEW

**Comment by Sealed101** [comment]
does the carpwave limit the dragon and the carptide to talking only in their channel like the blobwave?

**Comment by exymian** [comment]
wow what a great pr

**Comment by IndieanaJones** [comment]
> does the carpwave limit the dragon and the carptide to talking only in their channel like the blobwave?

As it is now, yes. Realistically, dragons and carps shouldn't be speaking galactic common (they're animals). If this bothers people enough I guess it could be looked into.

**Comment by MrDoomBringer** [comment]
> Space Dragon's fire breath no longer harms carp, but instead gives them an 8-second speed boost, equivalent to the one Space Dragon gets when enraged. Cold-blooded and everything, it makes sense.

Would be cool if this was moved to a seperate attack, maybe a battle-cry roar that affects nearby carp. I can otherwise see this causing a lot of confusion as carp might instinctively avoid the fire, dragons might not realize this is a feature (even if you do add a tooltip/desc to the action), etc.

**Comment by MrDoomBringer** [comment]
Otherwise really really cool PR

**Comment by DaydreamIQ** [comment]
Not sure if its worth addressing in this PR or not but one major issue I've noticed with dragon is that you can straight up two shot the rift with Hulk. Might be worth looking into if you're trying to make dragon more fun to play?

**Review by MrMelbert — COMMENTED** [review]
Just suggestions but bear in mind balloon alerts are typically concise, uncapitalized, and unpunctuated (except for failures)

File: code/modules/mob/living/simple_animal/hostile/space_dragon.dm [hunk 1]
```diff
@@ -178,6 +178,19 @@
 	add_dragon_overlay()
 	RegisterSignal(small_sprite, COMSIG_ACTION_TRIGGER, .proc/add_dragon_overlay)
 
+/mob/living/simple_animal/hostile/space_dragon/say(message, bubble_type, list/spans = list(), sanitize = TRUE, datum/language/language = null, ignore_spam = FALSE, forced = null, filterproof = null, message_range = 7, datum/saymode/saymode = null)
```
[the remark this answers is not in the document]
Comment (IndieanaJones) [reply code/modules/mob/living/simple_animal/hostile/space_dragon.dm:181]: I could look into seeing how aliens do their private channel.

**Comment by ATH1909** [comment]
> > does the carpwave limit the dragon and the carptide to talking only in their channel like the blobwave?
> 
> As it is now, yes. Realistically, dragons and carps shouldn't be speaking galactic common (they're animals). If this bothers people enough I guess it could be looked into.

space dragons should at least be able to speak Draconic, no?

**Comment by MrDoomBringer** [comment]
>  Realistically, dragons and carps shouldn't be speaking galactic common (they're animals)

If I wanted realism I'd go outside! Wordlessly killing people in this game is one of the lamest things you can do, anyone will tell you it's more fun to chest-beat, regardless of RP level. Ridiculous to try and force silent interactions for a fast-paced antagonist like dragon, where a curator acting as translator isn't likely. 

We can say the dragon picked up galcom by carefully listening to the screams of the crew they ate.

**Comment by IndieanaJones** [comment]
Yeah I'm just going to look into giving them a channel like the aliens do, not sure how stupid that code is. Worse case scenario I check for a :a at the beginning of their message and then do the channel thing if it finds it.

**Comment by bluetshirtguy** [comment]
cold blooded in space hmmmmm, space carps being COLD blooded in the insanely cold vacuum of space....

**Comment by MrDoomBringer** [comment]
>insanely cold vacuum of space

what

**Comment by gitbirb** [comment]
Maybe buff their healing? After all, what's the point in a large and powerful health pool if you cant maintain it no matter how hard you try. IMO the least fun part of space dragon is where your trying to heal, but cant because bodies only heal like 5% of your health and it takes like 5 minutes to heal to a combat-ready amount of health on a carp rift. They can't exactly go around looking for bodies either, since that'd leave their rift ungaurded, and can't exactly stand on their rift for 5 minutes to heal to full, since they'd just get shot at with 9000 lasers. 

Also maybe make carp rifts rise in percent faster (or buff carp rift sturdiness in general). If only 5 predetermined locations, people will probably learn and look for those places when they hear the announcement. Space dragons already struggle holding the rift when it's announced at 70% to the whole crew, so if crew knows at 0% (because of metaknowledge) they are gonna have a way more difficult time, especially with the low health pool of the rift.

**Comment by itsmeow** [comment]
Playing this downstream I also found that healing was incredibly hard a space dragon and required eating a LOT. I would definitely buff the healing at least double

**Comment by Mothblocks** [comment]
As we have an abnormally large number of open PRs, I am closing all ones that have been marked stale. Please find an interested maintainer over Discord to reopen and merge the PR.

**Comment by SethLafuente** [comment]
noooo this was a godo one

**Comment by thgvr** [comment]
re-open when

**Comment by IndieanaJones** [comment]
Will re-do this PR once I can assess how much Basic Mob Carps changes how Space Dragon plays.

**Comment by itsmeow** [comment]
Here is my wavespeak solution.

[0dde6a503e0d4c82e170a195561a0585e323d069](https://github.com/BeeStation/BeeStation-Hornet/pull/7500/commits/0dde6a503e0d4c82e170a195561a0585e323d069)

**Pull request closed without merging**